Source/Core/Common/MemArenaAndroid.cpp

```diff
@@ -96,12 +96,13 @@ u8* MemArena::ReserveMemoryRegion(size_t memory_size)
     PanicAlertFmt("Failed to map enough memory space: {}", LastStrerrorString());
     return nullptr;
   }
-  munmap(base, memory_size);
+  m_reserved_region = base;
   return static_cast<u8*>(base);
 }
 
 void MemArena::ReleaseMemoryRegion()
 {
+  munmap(m_reserved_region, m_reserved_region_size);
```

Comment:
I don't think `m_reserved_region_size` is set correctly on Android. You probably have to set it in ReserveMemoryRegion() like in the Unix variant. In fact, you can probably match both of those functions to the Unix ones (except for the flags):

https://github.com/dolphin-emu/dolphin/blob/8c67083c87e9b1235b3dd4d55aaec68ce4d5276a/Source/Core/Common/MemArenaUnix.cpp#L66-L87

Reply:
(Hm looking at this, does Android still have the old memory mapping bug where other malloc/new calls can allocate in fastmem space? Maybe that should be fixed too, but probably not in this PR.)

Reply:
Nobody ever ported the fix for that over to Android as far as I can recall. But yeah, let's leave it out of this PR.

Reply:
i did these changes because otherwise the fastmem arena and the lookup table would overlap and destroy each other.
but i can remove these changes from here once #11823 is merged (which i assume fixes it too)